Code/Java/CircularQueue

```diff
@@ -0,0 +1,402 @@
+
+/**
+ * Author - Khushee Kapoor
+ * 
+ * Ciruclar Queue is a Data Structure which operates on the FIFO (First In First Out) Principle.
+ * The pointers to the first and last element are dynamic. 
+ *                               
+ *          First element -> O - O                   Last element -> O - O <- First element
+ *                          /     \                                 /     \
+ *          Last element -> O     O      may also be                O     O
+ *                          \     /                                 \     /
+ *                           O - O                                   O - O 
+ * 
+ * 
+ *       O - O - O - O - O - O           may also be         O - O - O - O - O - O
+ *       ^                   ^                               ^   ^
+ *      first               last                           last first
+ * 
+ * In a normal Queue, we can insert elements until queue becomes full. But once queue becomes full, 
+ * we can not insert the next element even if there is a space in front of queue. This is possible
+ * in Circular Queues and hence is a optimizes space consumption.
+ */
+
+import java.util.*;
+public class CircularQueue
+{
+    int cq[];
+    int first, last;
+
+    CircularQueue(int size)
+    { 
+        //This is a default constructor to declare Circular Queue and global parameters first and last.
+        cq = new int[size];
+        first =-1;
+        last = -1;
+    }
+
+    public void insert(int value)
+    {
+        /**
```

Comment:
Add these in form of single line comments in between the code wherever required, Only add necessary comments.

Reply:
Made the changes.